messaging/src/main/java/org/axonframework/tracing/NoOpSpanFactory.java

```diff
@@ -37,30 +37,30 @@ public class NoOpSpanFactory implements SpanFactory {
 
     @Override
     public Span createRootTrace(Supplier<String> operationNameSupplier) {
-        return new NoOpSpan();
+        return NoOpSpan.INSTANCE;
```

Comment:
Note: I added this during experimentation with the NoOp versions we discussed on Slack. However, I then realised this would be better for performance. I'd like to keep it in here if you are okay with that.

Reply:
Smart!